### DISCUSSION TITLE
I wonder why it returns undefined when using useQuery if the value has not been loaded yet.

### DISCUSSION THREAD

**Discussion by Ubinquitous** [TanStack/query#7851] [Q&A]
I wonder why it returns undefined when using useQuery if the value has not been loaded yet.
I am curious as to why undefined is returned when using useQuery when the value has not yet been loaded.
Is there a reason not to have the code run synchronously until the value is not loaded?

**Comment by TkDodo** [comment]
> Is there a reason not to have the code run synchronously until the value is not loaded?

because it's not synchronous - it's asynchronous. What else should we do if we have to wait some time until data arrives over the network?

you might want to look into `suspense` though - it's react's way to make async code feel synchronous, and react query supports it with `useSuspenseQuery`. There, `data` won't be `undefined`.
